src/core.jl

```diff
@@ -282,6 +282,45 @@ Base.similar(A::AxisArray, ::Type{S}, ax1::Axis, axs::Axis...) where {S} = simil
     end
 end
 
+Base.convert(::Type{AxisArray{T, N, A, As}},
+             array::AxisArray{T, N, A, As}) where {T, N, A, As} = array
+Base.convert(::Type{AxisArray{TT, N, AA, As}},
+             array::AxisArray{T, N, A, As}) where {TT, N, AA, As, T, A} =
+    AxisArray(convert(AA, array), axes(array))
+function Base.convert(I::Type{AxisArray{TT, N, AA, AAs}},
+                      array::AxisArray{T, N, A, As}
+                     )::AxisArray{TT, N, AA, AAs} where {TT, N, AA, AAs, T, A, As}
+    perm = map(x -> findfirst(y -> typeof(y) <: x, axes(array)),
+               I.parameters[end].parameters)
+    any(perm .== 0) && throw(ArgumentError("Axes do not match"))
+    AxisArray(permutedims(convert(AA, array), perm), axes(array)[perm])
+end
+
+function Base.copy!(dest::AxisArray{TT, N, AA, As},
+                    source::AxisArray{T, N, A, As}) where {TT, N, AA, T, A, As}
+    copy!(dest.data, source.data)
+    dest
+end
+
+""" Copies data from one AxisArray to another.
+
+The axes should be the same (name and length) to a permutation. Alternatively, the sizes of
+the two arrays should match.
+"""
+Base.copy!(dest::AxisArray{TT, N, AA, AAs},
+           source::AxisArray{T, N, A, As}) where {TT, N, AA, AAs, T, A, As} = begin
+    perm = map(x -> findfirst(y -> y == x, axes(source)), axes(dest))
```

Comment:
This isn't going to check if the axes are the same up to name and length as the docstring says. Rather, the axis scaling is also compared, which you can see here:

```
julia> Axis{:a}(1:3) == Axis{:a}(2:4)
false
```